realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -263,33 +284,24 @@ public static void validateTable(ImplicitTransaction transaction) {
             if (!transaction.hasTable("class_AllTypes")) {
                 throw new RealmMigrationNeededException(transaction.getPath(), "Missing class 'class_AllTypes' for field 'columnObject'");
             }
-            if(!columnTypes.containsKey("columnRealmList")) {
+
+            Table table_7 = transaction.getTable("class_AllTypes");
+            if (!table.getLinkTarget(INDEX_COLUMNOBJECT).equals(table_7)) {
+                throw new RealmMigrationNeededException(transaction.getPath(), "Mismatching link tables for field 'columnObject'");
```

Comment:
Very core centric error msg. Perhaps "Mismatching field type for field 'xxx'...Even better would be "Mismatching field type for field 'xxx'. Was y expected z". y and z probably has the format "class_Y" and "class_Z" but that is ok I think